diff --git a/src/peer-routing.js b/src/peer-routing.js
index f37ce6638fabe3b31161ea2108be2e8caadafb3e..08d8b482dfbe285c20f14239837e64d00dbfd92e 100644
--- a/src/peer-routing.js
+++ b/src/peer-routing.js
@@ -111,7 +111,13 @@ class PeerRouting {
 
     const output = await pipe(
       merge(
-        ...this._routers.map(router => [router.findPeer(id, options)])
+        ...this._routers.map(router => (async function* () {
+          try {
+            yield [await router.findPeer(id, options)];
+          } catch (err) {
+            yield;
+          }
+        })())
       ),
       (source) => filter(source, Boolean),
       // @ts-ignore findPeer resolves a Promise
