error[E0446]: private type `PrivateEnum` in public interface
 --> tests/ui/pin_project/private_in_public-enum.rs:6:7
  |
6 |     V(PrivateEnum), //~ ERROR E0446
  |       ^^^^^^^^^^^ can't leak private type
...
9 | enum PrivateEnum {
  | ---------------- `PrivateEnum` declared as private

error[E0446]: private type `foo::PrivateEnum` in public interface
  --> tests/ui/pin_project/private_in_public-enum.rs:15:11
   |
15 |         V(PrivateEnum), //~ ERROR E0446
   |           ^^^^^^^^^^^ can't leak private type
...
18 |     enum PrivateEnum {
   |     ---------------- `foo::PrivateEnum` declared as private
