error: #[pin_project] attribute has been removed
  --> tests/ui/pin_project/remove-attr-from-struct.rs:22:1
   |
22 | #[pin_project] //~ ERROR has been removed
   | ^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `::pin_project::__private::__PinProjectInternalDerive` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot find attribute `pin` in this scope
  --> tests/ui/pin_project/remove-attr-from-struct.rs:18:7
   |
18 |     #[pin] //~ ERROR cannot find attribute `pin` in this scope
   |       ^^^

error: cannot find attribute `pin` in this scope
  --> tests/ui/pin_project/remove-attr-from-struct.rs:11:7
   |
11 |     #[pin] //~ ERROR cannot find attribute `pin` in this scope
   |       ^^^

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/remove-attr-from-struct.rs:35:16
   |
35 |     is_unpin::<A>(); //~ ERROR E0277
   |                ^ within `A`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `A`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:10:8
   |
10 | struct A {
   |        ^
note: required by a bound in `is_unpin`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:6:16
   |
6  | fn is_unpin<T: Unpin>() {}
   |                ^^^^^ required by this bound in `is_unpin`

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/remove-attr-from-struct.rs:36:16
   |
36 |     is_unpin::<B>(); //~ ERROR E0277
   |                ^ within `B`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `B`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:17:8
   |
17 | struct B {
   |        ^
note: required by a bound in `is_unpin`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:6:16
   |
6  | fn is_unpin<T: Unpin>() {}
   |                ^^^^^ required by this bound in `is_unpin`

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/remove-attr-from-struct.rs:40:22
   |
40 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |             -------- ^^^^^^ within `A`, the trait `Unpin` is not implemented for `PhantomPinned`
   |             |
   |             required by a bound introduced by this call
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `A`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:10:8
   |
10 | struct A {
   |        ^
note: required by a bound in `Pin::<P>::new`
  --> $RUST/core/src/pin.rs
   |
   | impl<P: Deref<Target: Unpin>> Pin<P> {
   |                       ^^^^^ required by this bound in `Pin::<P>::new`
...
   |     pub const fn new(pointer: P) -> Pin<P> {
   |                  --- required by a bound in this associated function

error[E0599]: no method named `project` found for struct `Pin<&mut A>` in the current scope
  --> tests/ui/pin_project/remove-attr-from-struct.rs:40:30
   |
40 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |                              ^^^^^^^ method not found in `Pin<&mut A>`

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/pin_project/remove-attr-from-struct.rs:43:22
   |
43 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |             -------- ^^^^^^ within `B`, the trait `Unpin` is not implemented for `PhantomPinned`
   |             |
   |             required by a bound introduced by this call
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `B`
  --> tests/ui/pin_project/remove-attr-from-struct.rs:17:8
   |
17 | struct B {
   |        ^
note: required by a bound in `Pin::<P>::new`
  --> $RUST/core/src/pin.rs
   |
   | impl<P: Deref<Target: Unpin>> Pin<P> {
   |                       ^^^^^ required by this bound in `Pin::<P>::new`
...
   |     pub const fn new(pointer: P) -> Pin<P> {
   |                  --- required by a bound in this associated function

error[E0599]: no method named `project` found for struct `Pin<&mut B>` in the current scope
  --> tests/ui/pin_project/remove-attr-from-struct.rs:43:30
   |
43 |     let _ = Pin::new(&mut x).project(); //~ ERROR E0277,E0599
   |                              ^^^^^^^ method not found in `Pin<&mut B>`
