error: no rules expected the token `(`
 --> tests/ui/pin_project/invalid.rs:5:14
  |
5 |         #[pin()] //~ ERROR no rules expected the token `(`
  |              ^ no rules expected this token in macro call
  |
note: while trying to match `]`
 --> src/lib.rs
  |
  |                 $(#[$pin:ident])?
  |                               ^

error: no rules expected the token `(`
 --> tests/ui/pin_project/invalid.rs:5:14
  |
5 |         #[pin()] //~ ERROR no rules expected the token `(`
  |              ^ no rules expected this token in macro call
  |
note: while trying to match `]`
 --> src/lib.rs
  |
  |                 $(#[$pin:ident])?
  |                               ^

error: no rules expected the token `#`
  --> tests/ui/pin_project/invalid.rs:20:9
   |
20 |         #[pin] //~ ERROR no rules expected the token `#`
   |         ^ no rules expected this token in macro call
   |
note: while trying to match meta-variable `$field_vis:vis`
  --> src/lib.rs
   |
   |                 $field_vis:vis $field:ident: $field_ty:ty
   |                 ^^^^^^^^^^^^^^

error: no rules expected the token `#`
  --> tests/ui/pin_project/invalid.rs:20:9
   |
20 |         #[pin] //~ ERROR no rules expected the token `#`
   |         ^ no rules expected this token in macro call
   |
note: while trying to match meta-variable `$field_vis:vis`
  --> src/lib.rs
   |
   |                 $field_vis:vis $field:ident: $field_ty:ty
   |                 ^^^^^^^^^^^^^^

error: cannot find attribute `pin` in this scope
  --> tests/ui/pin_project/invalid.rs:11:7
   |
11 |     #[pin] //~ ERROR cannot find attribute `pin` in this scope
   |       ^^^
