error: no rules expected the token `}`
 --> tests/ui/pin_project/unsupported.rs:3:1
  |
3 | / pin_project! {
4 | |     struct Struct1 {} //~ ERROR no rules expected the token `}`
5 | | }
  | |_^ no rules expected this token in macro call
  |
note: while trying to match meta-variable `$field_vis:vis`
 --> src/lib.rs
  |
  |                 $field_vis:vis $field:ident: $field_ty:ty
  |                 ^^^^^^^^^^^^^^
  = note: this error originates in the macro `$crate::__pin_project_expand` which comes from the expansion of the macro `pin_project` (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `}`
 --> tests/ui/pin_project/unsupported.rs:3:1
  |
3 | / pin_project! {
4 | |     struct Struct1 {} //~ ERROR no rules expected the token `}`
5 | | }
  | |_^ no rules expected this token in macro call
  |
note: while trying to match meta-variable `$field_vis:vis`
 --> src/lib.rs
  |
  |                 $field_vis:vis $field:ident: $field_ty:ty
  |                 ^^^^^^^^^^^^^^
  = note: this error originates in the macro `$crate::__pin_project_expand` which comes from the expansion of the macro `pin_project` (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `(`
 --> tests/ui/pin_project/unsupported.rs:8:19
  |
8 |     struct Struct2(); //~ ERROR no rules expected the token `(`
  |                   ^ no rules expected this token in macro call
  |
note: while trying to match `{`
 --> src/lib.rs
  |
  |         {
  |         ^

error: no rules expected the token `;`
  --> tests/ui/pin_project/unsupported.rs:12:19
   |
12 |     struct Struct3; //~ ERROR no rules expected the token `;`
   |                   ^ no rules expected this token in macro call
   |
note: while trying to match `{`
  --> src/lib.rs
   |
   |         {
   |         ^

error: no rules expected the token `(`
  --> tests/ui/pin_project/unsupported.rs:17:10
   |
17 |         A(u8)
   |          ^ no rules expected this token in macro call
   |
note: while trying to match `}`
  --> src/lib.rs
   |
   |         }
   |         ^

error: no rules expected the token `(`
  --> tests/ui/pin_project/unsupported.rs:17:10
   |
17 |         A(u8)
   |          ^ no rules expected this token in macro call
   |
note: while trying to match `}`
  --> src/lib.rs
   |
   |         }
   |         ^

error: no rules expected the token `union`
  --> tests/ui/pin_project/unsupported.rs:21:1
   |
21 | / pin_project! {
22 | |     union Union { //~ ERROR no rules expected the token `union`
23 | |         x: u8,
24 | |     }
25 | | }
   | |_^ no rules expected this token in macro call
   |
note: while trying to match `struct`
  --> src/lib.rs
   |
   |         [$(#[$attrs:meta])* $vis:vis struct $ident:ident]
   |                                      ^^^^^^
   = note: captured metavariables except for `:tt`, `:ident` and `:lifetime` cannot be compared to other tokens
   = note: see <https://doc.rust-lang.org/nightly/reference/macros-by-example.html#forwarding-a-matched-fragment> for more information
   = note: this error originates in the macro `$crate::__pin_project_expand` which comes from the expansion of the macro `pin_project` (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `union`
  --> tests/ui/pin_project/unsupported.rs:21:1
   |
21 | / pin_project! {
22 | |     union Union { //~ ERROR no rules expected the token `union`
23 | |         x: u8,
24 | |     }
25 | | }
   | |_^ no rules expected this token in macro call
   |
note: while trying to match `struct`
  --> src/lib.rs
   |
   |         [$(#[$attrs:meta])* $vis:vis struct $ident:ident]
   |                                      ^^^^^^
   = note: captured metavariables except for `:tt`, `:ident` and `:lifetime` cannot be compared to other tokens
   = note: see <https://doc.rust-lang.org/nightly/reference/macros-by-example.html#forwarding-a-matched-fragment> for more information
   = note: this error originates in the macro `$crate::__pin_project_expand` which comes from the expansion of the macro `pin_project` (in Nightly builds, run with -Z macro-backtrace for more info)
