error[E0277]: `PhantomPinned` cannot be unpinned
 --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:8:28
  |
8 |     impl Unpin for A where PhantomPinned: Unpin {} //~ ERROR E0277
  |                            ^^^^^^^^^^^^^^^^^^^^ the trait `Unpin` is not implemented for `PhantomPinned`
  |
  = note: consider using the `pin!` macro
          consider using `Box::pin` if you need to access the pinned value outside of the current scope
  = help: see issue #48214
  = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:16:28
   |
16 |     impl Unpin for B where Wrapper<PhantomPinned>: Unpin {} //~ ERROR E0277
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required for `phantom_pinned::Wrapper<PhantomPinned>` to implement `Unpin`
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:12:13
   |
12 |     impl<T> Unpin for Wrapper<T> where T: Unpin {}
   |             ^^^^^     ^^^^^^^^^^          ----- unsatisfied trait bound introduced here
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:34:28
   |
34 |     impl Unpin for A where Inner: Unpin {} //~ ERROR E0277
   |                            ^^^^^^^^^^^^ within `Inner`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `Inner`
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:30:12
   |
30 |     struct Inner(PhantomPinned);
   |            ^^^^^
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: `PhantomPinned` cannot be unpinned
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:42:28
   |
42 |     impl Unpin for B where Wrapper<Inner>: Unpin {} //~ ERROR E0277
   |                            ^^^^^^^^^^^^^^^^^^^^^ within `Inner`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using the `pin!` macro
           consider using `Box::pin` if you need to access the pinned value outside of the current scope
note: required because it appears within the type `Inner`
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:30:12
   |
30 |     struct Inner(PhantomPinned);
   |            ^^^^^
note: required for `inner::Wrapper<Inner>` to implement `Unpin`
  --> tests/ui/unstable-features/trivial_bounds-feature-gate.rs:38:13
   |
38 |     impl<T> Unpin for Wrapper<T> where T: Unpin {}
   |             ^^^^^     ^^^^^^^^^^          ----- unsatisfied trait bound introduced here
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
